src/ddspdrum/module.py

```diff
@@ -348,22 +430,20 @@ def __init__(
         control_rate: int = CONTROL_RATE,
     ):
         super().__init__(sample_rate=sample_rate, control_rate=control_rate)
-
-        assert 0 <= ratio <= 1
-        self.ratio = ratio
+        self.add_parameter("ratio", ratio, 0, 1)
 
     def __call__(self, audio_in: np.ndarray):
         noise = self.noise_of_length(audio_in)
-        return crossfade(audio_in, noise, self.ratio)
+        return crossfade(audio_in, noise, self.parameters['ratio'].value)
 
     @staticmethod
     def noise_of_length(audio_in: np.ndarray):
         return np.random.rand(len(audio_in))
 
 
-class Synth:
+class Synth(SynthModule):
```

Comment:
This might be controversial -- I found I was duplicating a lot of the parameter updating code I wrote for the SynthModule to here, so thought that this could just be a SynthModule and use the same parameter logic as above. Easy to create named parameters that control underlying SynthModules.

Reply:
Big fan of this. Nice work/

Reply:
So this makes me slightly nervous, but I can live with it.

If the only reason for this is `sustain_duration` and `vco_1_ratio`, another approach would be:
* `sustain_duration` is a considered a meta-parameter, not a parameter. This is my putting on my ML hat. You would fix the sustain_duration when doing traditional ML. (However, there are more radical things where you might want to compare audio with different lengths. However, that might happen outside this anyway, and for mini-batching you'd want a fixed sustain_duration so each audio is the same length.)
* `vco_1_ratio` is part of a dummy Module that contains just a parameter that can be queried, but doesn't do any audio processing.

Reply:
Alternately, each Synth contains a ParametersModule which is just parameters.

Reply:
It wasn't just for the `sustain_duration` and `vco_1_ratio` -- also to handle the nested parameters within the SynthModules. I kind of like thinking about a Synth as just a SynthModule that contains nested SynthModules, could lead to some interesting architectures. That being said, with the comments on how to remove connect_parameters and the idea of having a ParametersModule, it would totally work to remove the SynthModule inheritance -- might be better to keep relationships simplified.

Reply:
I think let's keep it flat for now, let's get a few more things nailed down, and then let's make some cool nested architectures soon

Reply:
I added this https://github.com/turian/ddsp-drum/pull/20 but see discussion we were having about how we can't make Synth not a SynthModule and remove `connect_parameter` without making Synth have its own business logic for compound parameter addressing.